### PR TITLE
If none of these environmental variables exist; this will crash.  

### PR DESCRIPTION
**Pull request by Nathanaela**
Create a workable fallback.

### CODE REVIEW

File: services/cancellation.ts [hunk 1]
```diff
@@ -58,7 +58,12 @@ class CancellationService implements ICancellationService {
 	}
 
 	private static get killSwitchDir(): string {
-		return path.join(os.tmpdir(), process.env.SUDO_USER || process.env.USER || process.env.USERNAME, "KillSwitches");
+        var tmp = process.env.SUDO_USER || process.env.USER || process.env.USERNAME;
+        if (tmp) {
+            return path.join(os.tmpdir(), tmp, "KillSwitches");
+        } else {
+            return path.join(os.tmpdir(), "KillSwitches");
+        }
```
Comment (teobugslayer) [inline services/cancellation.ts]: Isn't this equivalent to
```JavaScript
var tmp = process.env.SUDO_USER || process.env.USER || process.env.USERNAME || "";
return path.join(os.tmpdir(), tmp, "KillSwitches");
```

**Comment by teobugslayer** [comment]
Hello @Nathanaela,
The issue with this code is that if you have machine with two simultaneously logged in users, it would either break due to missing write permissions, or one user would stop the running CLI process of the other user. What could be done is such case?

File: services/cancellation.ts (hunk 1 above)
Comment (Nathanaela) [inline services/cancellation.ts]: Interesting.   I just tested path.join and your understanding is correct.  I had expected path join to do two slashes in the middle of the path...   I'll repush a commit.

**Comment by Nathanaela** [comment]
Are you sure it isn't running.  I just installed NativeScript yesterday and this code file is present on my box...

**Comment by teobugslayer** [comment]
@Nathanaela,

the code is instantiated only on Windows - see https://github.com/telerik/mobile-cli-lib/commit/d1230fadf8f3dbabde8593662100cf3912e07e95#diff-dc9cf03aafbce6404bc646fee9d733c0R82

You are correct that this code can potentially fail on Windows, though I believe it is highly unlike for USERNAME to miss on Windows.

Maybe the correct fix is:

```
private static get killSwitchDir(): string {
		return path.join(os.tmpdir(), process.env.USERNAME || "", "KillSwitches");
	}
```

**Comment by Nathanaela** [comment]
It is unlikely that it would now fail on windows; as username is set by the os and someone would have to actively remove it from the environment.    But it might be worth doing the patch so that in the future if it is changed to be active on linux/osx it won't fail again.

**Comment by teobugslayer** [comment]
I agree. Thank you for your effort!

**Pull request merged**